libvips/include/vips/foreign.h

```diff
@@ -371,19 +371,21 @@ int vips_openslideload_source( VipsSource *source, VipsImage **out, ... )
 	__attribute__((sentinel));
 
 /**
- * VipsForeignJpegSubsample:
- * @VIPS_FOREIGN_JPEG_SUBSAMPLE_AUTO: default preset
- * @VIPS_FOREIGN_JPEG_SUBSAMPLE_ON: always perform subsampling
- * @VIPS_FOREIGN_JPEG_SUBSAMPLE_OFF: never perform subsampling
+ * VipsForeignSubsample:
+ * @VIPS_FOREIGN_SUBSAMPLE_AUTO: prevent subsampling when quality > 90
+ * @VIPS_FOREIGN_SUBSAMPLE_ON: always perform subsampling
+ * @VIPS_FOREIGN_SUBSAMPLE_OFF: never perform subsampling
  *
- * Set jpeg subsampling mode.
+ * Set subsampling mode.
  */
 typedef enum {
-	VIPS_FOREIGN_JPEG_SUBSAMPLE_AUTO,
-	VIPS_FOREIGN_JPEG_SUBSAMPLE_ON,
-	VIPS_FOREIGN_JPEG_SUBSAMPLE_OFF,
-	VIPS_FOREIGN_JPEG_SUBSAMPLE_LAST
-} VipsForeignJpegSubsample;
+	VIPS_FOREIGN_SUBSAMPLE_AUTO,
+	VIPS_FOREIGN_SUBSAMPLE_ON,
+	VIPS_FOREIGN_SUBSAMPLE_OFF,
+	VIPS_FOREIGN_SUBSAMPLE_LAST
+} VipsForeignSubsample;
+
+typedef VipsForeignSubsample VipsForeignJpegSubsample;
```

Comment:
Is this enough for backwards compatibility?

Reply:
I was worrying about this in bed last night. Some bindings generate enums from introspection, and this change will prevent the old `VipsForeignJpegSubsample` enum being created. That in turn will cause an API break :+1: 

I think we'll need to keep both enums, but only use `VipsForeignSubsample` and add a note to the doc comment for `VipsForeignJpegSubsample` that it's deprecated and that people should use `VipsForeignSubsample` in new code.